holoviews/core/spaces.py

```diff
@@ -1054,6 +1054,35 @@ def dynamic_hist(obj):
             return hist
 
 
+    def reindex(self, kdims=[], force=False):
+        """
+        Reindexing a DynamicMap allows reordering the dimensions but
+        not dropping an individual dimension. The force argument which
+        usually allows dropping non-constant dimensions is therefore
+        ignored and only for API consistency.
+        """
+        kdims = [self.get_dimension(kd, strict=True) for kd in kdims]
+        dropped = [kd for kd in self.kdims if kd not in kdims]
+        if dropped:
+            raise ValueError("DynamicMap does not allow dropping dimensions, "
+                             "reindex may only be used to reorder dimensions.")
+        reindexed = super(DynamicMap, self).reindex(kdims, force)
+        def reindex(*args, **kwargs):
+            keymap = {kd.name: arg for kd, arg in zip(self.kdims, args)}
+            keymap.update(kwargs)
```

Comment:
You mean you are assuming ``Callable`` can be invoked with a bunch of kwargs and nothing else?

That is fine unless the callback contains ``*args`` in which case ``Callable`` needs positional arguments it passes through directly to the callback. Once my PR is merged you can easily inspect the argspec to see if ``varargs`` is set, in which case you could raise another exception.